src/variables.jl

```diff
@@ -33,3 +33,72 @@ A [`AddVariableNotAllowed`](@ref) error is thrown if adding variables cannot be
 done in the current state of the model `model`.
 """
 add_variable(model::ModelLike) = throw(AddVariableNotAllowed())
+
+"""
+    add_constrained_variable(
+        model::ModelLike,
+        set::AbstractScalarSet
+    )::Tuple{MOI.VariableIndex,
+             MOI.ConstraintIndex{MOI.SingleVariable, typeof(set)}}
+
+Add to `model` a scalar variable constrained to belong to `set`, returning the
+index of the variable created and the index of the constraint constraining the
+variable to belong to `set`.
+
+By default, this function falls back to creating a free variable with
+[`add_variable`](@ref) and then constraining it to belong to `set` with
+[`add_constraint`](@ref), which turns the free variable into a constrained
```

Comment:
Is this description still accurate given the new definition of constrained variables?